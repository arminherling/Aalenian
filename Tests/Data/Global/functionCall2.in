define inner() 
{ 
}

define outer()
{
    inner()
}
