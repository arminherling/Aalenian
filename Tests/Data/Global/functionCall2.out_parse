FunctionDefinitionStatement: {
    NameExpression: inner
    ParametersNode(0): {
    }
    BlockNode(0): {
    }
}
FunctionDefinitionStatement: {
    NameExpression: outer
    ParametersNode(0): {
    }
    BlockNode(1): {
        ExpressionStatement: {
            FunctionCallExpression: {
                NameExpression: inner
                ArgumentsNode(0): {
                }
            }
        }
    }
}
