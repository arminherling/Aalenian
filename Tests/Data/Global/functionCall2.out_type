TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->()
    Name: inner
    Parameters(0): {
    }
    ReturnType: {
        Type: void
    }
    Body(0): {
    }
}
TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->()
    Name: outer
    Parameters(0): {
    }
    ReturnType: {
        Type: void
    }
    Body(1): {
        TypedExpressionStatement: {
            Type: void
            Expression: {
                TypedFunctionCallExpression: {
                    Type: void
                    Name: inner
                    ArgumentsNode(0): {
                    }
                }
            }
        }
    }
}
