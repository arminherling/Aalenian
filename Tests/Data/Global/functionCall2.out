FunctionDefinitionStatement: {
    Name: inner
    Parameters(0): {
    }
    Block(0): {
    }
}
FunctionDefinitionStatement: {
    Name: outer
    Parameters(0): {
    }
    Block(1): {
        ExpressionStatement: {
            FunctionCall: {
                Name: inner
                Arguments(0): {
                }
            }
        }
    }
}
